### ISSUE TITLE
Conflicting Jupyter kernel specs from multiple environments

### ISSUE THREAD

**Issue by bhrutledge** [django-extensions/django-extensions#1026]
I'm developing multiple Django projects, each with its own virtualenv that includes django-extensions. To user the Jupyter Notebook, I've been doing `pip install jupyter` followed by `manage.py shell_plus --notebook` inside the virtualenv. It's been working okay, but I've learned a little bit more about how Jupyter works, and it seems that the way django-extensions handles kernel specs is not ideal.

For example, when I run the notebook in `project-foo`, I get the following kernel spec at `/Users/brian/Library/Jupyter/kernels/django_extensions/kernel.json`:

```
{"language": "python", "display_name": "Django Shell-Plus", "env": {"PYTHONPATH": "/Users/brian/Code/project-foo"}, "argv": ["/Users/brian/.virtualenvs/project-foo/bin/python", "-m", "ipykernel", "-f", "{connection_file}", "--ext", "django_extensions.management.notebook_extension"]}
```

However, when I run the notebook in `project-bar`, that overwrites the *same file* with:

```
{"language": "python", "display_name": "Django Shell-Plus", "env": {"PYTHONPATH": "/Users/brian/Code/project-bar"}, "argv": ["/Users/brian/.virtualenvs/project-bar/bin/python", "-m", "ipykernel", "-f", "{connection_file}", "--ext", "django_extensions.management.notebook_extension"]}
```

Now, in practice, this hasn't been a huge problem, because I'm usually not running multiple notebook servers at once, and django-extensions always writes the kernel spec before it starts a notebook server. But it does seem like it would cause unexpected and undesirable behavior.

After reading up on [installing kernels](https://ipython.readthedocs.io/en/latest/install/kernel_install.html), I see two potential fixes:

* Instead of installing to the [user's kernel directory](https://github.com/django-extensions/django-extensions/blob/1.7.6/django_extensions/management/commands/shell_plus.py#L195), install to `sys.prefix`. IE, mimic `python -m ipykernel install --sys-prefix` instead of `--user`. So, in the example above, the kernel spec for `project-foo` would be installed to `/Users/brian/.virtualenvs/project-foo/share/jupyter/kernels/django-extensions`.
* Mimic the behavior of the `--name` option to `python -m ipykernel install`, similar to the [option for setting `display_name`](https://github.com/django-extensions/django-extensions/blob/1.7.6/django_extensions/management/commands/shell_plus.py#L224-L225). So, in the example above, the kernel spec for `project-foo` would be installed to `/Users/brian/Library/Jupyter/kernels/django_extensions-project-foo` (possibly with a `display_name` of `Django Shell-Plus (project-foo)`.

**Comment by trbs** [comment]
Thanks for the report and spending your time diving into this !

Happy with any proposal and will accept any PR to improve how we handle notebooks 👍

**Comment by bhrutledge** [comment]
I'm definitely game to work on this. Just gotta carve out the time.

**Comment by bhrutledge** [comment]
This is still on my radar, but not a high priority.

Copied from a [previous discussion](https://github.com/django-extensions/django-extensions/pull/1120#issuecomment-342686339):

What I'm planning to do is something like what I [suggested on #1104](https://github.com/django-extensions/django-extensions/issues/1104#issuecomment-341847301). More specifically, I'd add a new command, maybe called `install_kernel`. This would behave and have options similar to [`python -m ipykernel install`](https://ipython.readthedocs.io/en/latest/install/kernel_install.html#kernels-for-different-environments)  and the [bash kernel](https://github.com/takluyver/bash_kernel) with respect to installing a kernel spec, but would add the `django_extensions` notebook extension. I'd try to have helpful defaults, like defaulting to `--user` with a name that reflects the active virtual environment.

With the kernel spec installed, users would be free to run `jupyter notebook`, `jupyter lab`, `jupyter qtconsole`, etc. in whatever environment they want (their Django virtualenv, a Jupyter virtualenv, Anaconda, etc.). And, `django_extensions` wouldn't need the `NOTEBOOK_ARGUMENTS` and `IPYTHON_ARGUMENTS` settings, because it's no longer acting as a wrapper around the `notebook` command, and users would be free to edit the installed kernel spec as needed.

We'd use documentation to introduce the new command and recommend a switch, then later deprecating `--notebook` and `--kernel`, and eventually removing them.

**Comment by terencehonles** [comment]
I believe the cause of this is the same as #1378 and is fixed by #1500 but also #1509
